### PR TITLE
JNA Windows Scaling

### LINKED ISSUE #31: Improved zoom on windows
Right now display scaling on Windows 10 causes poor automatic zooming through Swing. This makes the interface appear pixelated and the custom swing elements (primarily JEditTextArea) do not calculate x positions for the caret correctly. After https://github.com/processing/processing4/pull/30, automatic swing zooming will be disabled. While this improves the experience for windows users that have display scaling enabled, this is a poor accessibility experience. I would like to re-enable automatic zoom on Windows and, ideally, extend zoom into native swing elements like the menu bar as well. However, as described in #30, we will need a new way to detect display scaling.

### PR DESCRIPTION
**Pull request by sampottinger**
Move to zoom level being calculated in the platform instead of DPI being returned. This is because zoom level may be distinct from the system DPI and operating systems may report a DPI of 96 while also requesting a display zooming of 125%. Therefore, using the system display scaling is preferred to using DPI after JEP 263. This also includes an attempted implementation of JNA based display scaling on windows. Resolves #31.

### CODE REVIEW

**Comment by sampottinger** [comment]
Thanks @Volts-s and @kllsamui for your help in testing! Removed the `[WIP]` and added @benfry.

File: app/src/processing/app/platform/DefaultPlatform.java [hunk 1]
```diff
@@ -157,8 +159,34 @@ public int unsetenv(String variable) {
...
+   * Note that this may be distinct from the system DPI and some operating
+   * systems may report a DPI of 96 while also requesting a display zooming of
+   * 125%. However, others may not report a "display scaling" but provide a
+   * DPI of 120 when the elements should have a 125% zoom. This will use the
+   * preferred method of determining the appropriate zoom given the platform
+   * in use. Using this system display scaling percentage approach instead of
+   * returning DPI directly is preferred after JEP 263.
+   * </p>
+   *
+   * @return The zoom level where 1 means 100% (no zoom) and 125% means 25%
```
Comment (benfry) [inline app/src/processing/app/platform/DefaultPlatform.java]: Should this read 1.0 means 100% and 1.25 means 125%?

**Comment by benfry** [comment]
Is this ready to merge?

File: app/src/processing/app/platform/DefaultPlatform.java (hunk 1 above)
Comment (sampottinger) [reply app/src/processing/app/platform/DefaultPlatform.java]: Whoops! Thanks for catching that. Yes that is correct.

Comment (sampottinger) [reply app/src/processing/app/platform/DefaultPlatform.java]: Resolved

**Comment by sampottinger** [comment]
Thanks for catching that @benfry. Yes, ready to merge when you are!

**Comment by benfry** [comment]
Great; thanks!

**Pull request merged**